### ISSUE TITLE
Ability to close immortal PRs forever

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#17985]
### What would you like Renovate to be able to do?

Add a way for for people insistent to ignore an "immortal" PR forever.

### If you have any ideas on how this should be implemented, please tell us here.

When PRs are "immortal", add a checkbox to the PR on supported platforms allowing the user to tick it to allow the PR to remain closed instead of being reopened.

UI wise I'm not sure where's best. Logic wise it should be similar to rebase/retry.

However, I'm not sure if we parse PR bodies for closed PRs. If not then it might be a lot of work (or too much in fact) and a different solution is necessary. Needs investigation.

### Is this a feature you are interested in implementing yourself?

No

**Comment by RahulGautamSingh** [comment]
Relevant issues:
https://github.com/renovatebot/config-help/issues/127
https://github.com/renovatebot/config-help/issues/409

**Comment by HonkingGoose** [comment]
## UI

> UI wise I'm not sure where's best. Logic wise it should be similar to rebase/retry.

Grouping the checkboxes is easiest, I think. Users already know where to find the _first_ checkbox, so it's logical to add a second checkbox.

I've taken a stab at mocking this up for us.

## Ideas to consider

Tell our users:

- that ignoring dependencies forever is not recommended by us
- to use the "Dependency Dashboard Approval" workflow instead of ignoring the dependency forever
- how to _recover_ from a bad click: "Help I clicked ignore by accident, how can I tell Renovate I want the update again?"

# PR Mockup (real markdown, not an image)

[![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Update | Change |
|---|---|
| lockFileMaintenance | All locks refreshed |

🔧 This Pull Request updates lock files to use the latest dependency versions.

---

### Configuration

📅 **Schedule**: Branch creation - "before 5am on monday" (UTC), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get [config help](https://togithub.com/renovatebot/renovate/discussions) if that's undesired.

---

 - [ ] If you want to rebase/retry this PR, click this checkbox. :warning: Warning: custom changes will be lost.
 
 - [ ] To ignore this PR forever, click this checkbox. :warning: We do _not_ recommend that you ignore PRs forever! Remind yourself to check this dependency later!

---

This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://app.renovatebot.com/dashboard#github/HonkingGoose/git-gosling).


# Markdown code for the mockup

```markdown
# PR Mockup

[![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Update | Change |
|---|---|
| lockFileMaintenance | All locks refreshed |

🔧 This Pull Request updates lock files to use the latest dependency versions.

---

### Configuration

📅 **Schedule**: Branch creation - "before 5am on monday" (UTC), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get [config help](https://togithub.com/renovatebot/renovate/discussions) if that's undesired.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, click this checkbox. :warning: Warning: custom changes will be lost.
 
 - [ ] <!-- close-immortal-pr -->If you want to ignore this PR forever, click this checkbox. :warning: We do _not_ recommend that you ignore PRs forever! Remind yourself to check this dependency later!

---

This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://app.renovatebot.com/dashboard#github/HonkingGoose/git-gosling).
<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzMi4xMjYuMiIsInVwZGF0ZWRJblZlciI6IjMyLjEyNi4yIn0=-->
```

**Comment by AWare** [comment]
This is really important because if you accidentally create a pr from the dependency dashboard, it's impossible to close it.

**Comment by neoeinstein** [comment]
Could this be implemented as something with a label? Such as `renovate:forbid`? Then, if that label were found on the pull/merge request, when Renovate would want to open up a PR with the same branch name as a forbidden, closed PR, it wouldn't.

**Comment by rarkins** [comment]
Yes, but please create a separate "Suggest an idea" discussion for that

**Comment by seanmc9** [comment]
> Yes, but please create a separate "Suggest an idea" discussion for that

went ahead and did this here https://github.com/renovatebot/renovate/discussions/29930 because this would be super helpful for us, particularly to avoid immortal PRs clogging our build pipeline by rebasing on every push to the default branch.
